src/plugins/data/public/search/aggs/buckets/terms_fn.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { ExpressionFunctionDefinition } from '../../../../../expressions/public';
+import { AggExpressionType, AggExpressionFunctionArgs } from '../';
+
+const aggTypeName = 'terms';
+const fnName = 'aggTypeTerms';
```

Comment:
Not sure if there is discussion somewhere about naming these agg type functions, but I believe we usually use snake case like `kibana_context` and `var_set`, maybe for consistency we could use that for agg types too?

Like

```
agg_type_terms
```

or maybe better, we can make it short

```
agg_terms
```

maybe we can use dot `.` (not sure if expression grammar allows that)

```
agg.terms
```

Reply:
> Not sure if there is discussion somewhere about naming these agg type functions, but I believe we usually use snake case

Yeah so I was trying to remember this because I know we've had the discussion before and I cannot find the issue...

I did camelCase here because [that's what all of the Canvas functions do](https://www.elastic.co/guide/en/kibana/current/canvas-function-reference.html), but I'm happy to go with snake_case if we want to move in that direction instead, as long as we stick to something.

I like the idea of removing `type` to simplify, I'll plan to do that once we determine the snake vs camel case question.

Reply:
> I did camelCase here because that's what all of the Canvas functions do...

Good point, so we are inconsistent with Canvas. But we don't have many of those snake case functions...

I wonder which pattern lends uses...

They use snake case: https://github.com/elastic/kibana/blob/master/x-pack/plugins/lens/public/xy_visualization/xy_expression.tsx#L70

Reply:
no preference but i agree we should be consistent.

Reply:
i've changed to camelCase `aggTerms` for now & pinged the relevant team members to discuss

Reply:
Decided today we would stick with camelCase for new functions moving forward, as a switch to snake_case would have heavy breaking change for saved Canvas expression functions.